sorc/ncep_post.fd/MISCLN.f

```diff
@@ -3736,6 +3752,328 @@ SUBROUTINE MISCLN
+                enddo
+              enddo
+             endif
+           ENDIF
+
+!Mixed Layer (100 mb) Virtual LFC
+
+           IF (IGET(992)>0) THEN
+!$omp parallel do private(i,j)
+           DO J=JSTA,JEND
+             DO I=1,IM
+               EGRID1(I,J) = -H99999
+               EGRID2(I,J) = -H99999
+               EGRID3(I,J) = -H99999
+               EGRID4(I,J) = -H99999
+               EGRID5(I,J) = -H99999
+               EGRID6(I,J) = -H99999
+               EGRID7(I,J) = -H99999
+               EGRID8(I,J) = -H99999
+               LB2(I,J)  = (LVLBND(I,J,1) + LVLBND(I,J,2) +           &
+                            LVLBND(I,J,3))/3
+               P1D(I,J)  = (PBND(I,J,1) + PBND(I,J,2) + PBND(I,J,3))/3
+               T1D(I,J)  = (TVIRTUAL(TBND(I,J,1),QBND(I,J,1)) +       &
+                            TVIRTUAL(TBND(I,J,2),QBND(I,J,2)) +       &
+                            TVIRTUAL(TBND(I,J,3),QBND(I,J,3)))/3
+               Q1D(I,J)  = (QBND(I,J,1) + QBND(I,J,2) + QBND(I,J,3))/3
+             ENDDO
+           ENDDO
+
+             DPBND = 0.
+             ITYPE = 2
+! EGRID3 is Virtual LFC
+             CALL CALCAPE2(ITYPE,DPBND,P1D,T1D,Q1D,LB2,            &
+                           EGRID1,EGRID2,EGRID3,EGRID4,EGRID5,     &
+                           EGRID6,EGRID7,EGRID8)
+
+             DO J=JSTA,JEND
+               DO I=1,IM
+                 GRID1(I,J) = EGRID3(I,J)
+               ENDDO
+             ENDDO
+             CALL BOUND(GRID1,D00,H99999)
+             if(grib=='grib2') then
+               cfld=cfld+1
+               fld_info(cfld)%ifld=IAVBLFLD(IGET(992))
+               fld_info(cfld)%lvl=LVLSXML(1,IGET(992))
+!$omp parallel do private(i,j,jj)
+               do j=1,jend-jsta+1
+                 jj = jsta+j-1
+                 do i=1,im
+                   datapd(i,j,cfld) = GRID1(i,jj)
+                 enddo
+               enddo
+             endif
+           ENDIF   !992
+
+
+           IF (IGET(763)>0) THEN
+!$omp parallel do private(i,j)
+! EGRID3 is Virtual LFC
+             DO J=JSTA,JEND
+               DO I=1,IM
+                 GRID1(I,J) = Q1D(I,J)
+               ENDDO
+             ENDDO
+             if(grib=='grib2') then
+               cfld=cfld+1
+               fld_info(cfld)%ifld=IAVBLFLD(IGET(763))
+               fld_info(cfld)%lvl=LVLSXML(1,IGET(763))
+!$omp parallel do private(i,j,jj)
+               do j=1,jend-jsta+1
+                 jj = jsta+j-1
+                 do i=1,im
+                   datapd(i,j,cfld) = GRID1(i,jj)
+                 enddo
+               enddo
+             endif
+           ENDIF  
+
+!Hail parameter
+            IF (IGET(993)>0) THEN
+            DO J=JSTA,JEND
+               DO I=1,IM
+!               LAPSE=-((T700(I,J)-T500(I,J))/(D50*(Z700(I,J)-Z500(I,J))))
+                LAPSE=(T700(I,J)-T500(I,J)) - 7.
+                if (LAPSE <= 0.)LAPSE=0.
+                SHIP=(MUCAPE(I,J)*D1000*Q1D(I,J)*LAPSE*(T500(I,J))*FSHR(I,J))/HCONST
```

Comment:
Hi Ed,
The Q1D  I mentioned in last comment is this Q1D used to compute SHIP. In last comment, I put the comment to the Q1S used to compute mixed layer (or LFC?)_, sorry for the confusion.
So in this code block to compute SHIP,  where does this Q1D come from? Since no CALCAPE or CALCAPE2 is called in this "IF (IGET(993)>0) THEN" block, so it has to come from somewhere before this IF-block, like just from the IF-block above for IGET(992) for mixed layer (or LFC?).
Thanks!
- Gang